wptherml/exciton.py

```diff
@@ -0,0 +1,594 @@
+        H_nm : float
+            The matrix elements corresponding to the interactions between sites n and m
+        """
+     
+        H_nm = self.exciton_energy * (n == m)
+        return H_nm
+
+    def _compute_dipole_dipole_coupling(self, n, m):
+        """Method to compute the dipole-dipole potential between excitons located on site n and site m
+
+        Arguments
+        ---------
+        n : int
+            the index of site n 
+        m : int
+            the index of site m 
+
+        Attributes
+        ----------
+        coords : 3 x number_of_monomers numpy array of floats
+            the cartesian coordinates of each monomer
+
+        transition_dipole_moment : 1x3 numpy array of floats
+            the transition dipole moment associated with the excitons
+
+        Returns
+        -------
+        V_nm : float
+             the dipole-dipole potential between exciton on site n and m
+        """
+
+        # calculate separation vector between site m and site n
+        _r_vec = self.coords[m, :] - self.coords[n, :]
+
+        # self.transition_dipole_moment is the transition dipole moment!
+        if n != m:
+            V_nm = (
+                1 / (self.refractive_index**2 * np.sqrt(np.dot(_r_vec, _r_vec)) ** 3)
+            ) * (
+                np.dot(self.transition_dipole_moment, self.transition_dipole_moment)
+                - 3
+                * (
+                    (
+                        np.dot(self.transition_dipole_moment, _r_vec)
+                        * np.dot(_r_vec, self.transition_dipole_moment)
+                    )
+                    / (np.sqrt(np.dot(_r_vec, _r_vec)) ** 2)
+                )
+            )
+        else:
+            V_nm = 0
+
+        return V_nm
+    
+    def _compute_2D_dd_coupling(self, r_vector):
+        """ Function that computes the dipole dipole coupling contribution of the total energy of a system based upon mu_d,
+            the transition dipole moment of the donor, mu_a, the transition dipole moment of the acceptor, r_vector, the 
+            distance separating the donor and acceptor, and the refractive index, a paremeter that describes the effect of 
+            the system on light.
+    
+        Arguments
+        ---------
+        mu_d : numpy array of floats
+         the transition dipole mooment of the donor
+        mu_a : numpy array of floats
+            the transition dipole moment of the acceptor
+        r : numpy array of floats
+            the distance separating the donor and acceptor
+        n : float
+            refractive index of the medium
+        """
+
+        r_scalar = np.sqrt(np.dot(r_vector, r_vector))
+    
+        return (1 / (self.refractive_index ** 2 * r_scalar ** 3)) * (np.dot(self.transition_dipole_moment, self.transition_dipole_moment) - 3 * (np.dot(self.transition_dipole_moment, r_vector) * np.dot(r_vector, self.transition_dipole_moment)) / r_scalar ** 2)
+
+    def build_exciton_hamiltonian(self):
```

Comment:
With the cartesian coordinates specified, this will compute the Hamiltonian 1D, 2D, or 3D aggregates.